### PR TITLE
WEBUI-1385: Fix update of Template properties

### PR DESCRIPTION
**Pull request by alokhyland**

### CODE REVIEW

**Review by swarnadipa-dev — DISMISSED** [review]

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js [hunk 1]
```diff
@@ -191,7 +192,8 @@ Polymer({
   _documentChanged() {
     if (this.document) {
       this.set('document.properties.tmpl:templateType', this.document.properties['tmpl:templateType'] || 'auto');
-      this.set('editedDocument', JSON.parse(JSON.stringify(this.document)));
+      this.set('editedDocument', this._parseJSON(this.document));
+      if (this.unmodifiedDocument === undefined) this.unmodifiedDocument = this._parseJSON(this.document);
```
Comment (madhurkulshrestha-hyland) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: Use ternary operator for this condition

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js [hunk 2]
```diff
@@ -175,6 +175,7 @@ Polymer({
       type: String,
       value: 'view',
     },
+    unmodifiedDocument: Object,
```
Comment (madhurkulshrestha-hyland) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: Change name to originalDocument

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js [hunk 3]
```diff
@@ -216,12 +218,33 @@ Polymer({
...
+
+  _parseJSON(obj) {
+    return JSON.parse(JSON.stringify(obj));
   },
 
   _save() {
-    delete this.editedDocument.properties['dc:contributors'];
-    return this.$.doc.put();
+    const modified = this._parseJSON(this.editedDocument.properties);
+    const org = this._parseJSON(this.unmodifiedDocument.properties);
```
Comment (madhurkulshrestha-hyland) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: Name should be meaningful. Change this to "orig" or "originalDocument"

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js (hunk 1 above)
Comment (alokhyland) [reply addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: got lint issue

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js (hunk 2 above)
Comment (alokhyland) [reply addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: done

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js (hunk 3 above)
Comment (alokhyland) [reply addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: done

**Review by madhurkulshrestha-hyland — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

File: addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js [hunk 4]
```diff
@@ -216,12 +218,33 @@ Polymer({
   },
 
   _resetConfig() {
-    this.set('editedDocument', JSON.parse(JSON.stringify(this.document)));
+    this.set('editedDocument', this._parseJSON(this.document));
+  },
+
+  _findChangedValues(object1, object2) {
```
Comment (swarnadipa-dev) [inline addons/nuxeo-template-rendering/elements/nuxeo-template-rendering-page.js]: please change object1, object2 parameter names to something that is understandable in context of the code.

**Review by swarnadipa-dev — APPROVED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Pull request merged**